packages/compat/src/compat-app.ts

```diff
@@ -233,23 +233,11 @@ class CompatAppAdapter implements AppAdapter<TreeNames> {
     // webpack's default is ['.wasm', '.mjs', '.js', '.json']. Keeping that
     // subset in that order is sensible, since many third-party libraries will
     // expect it to work that way.
-    let extensions = ['.wasm', '.mjs', '.js', '.json', '.hbs'];
-
-    // for now, this is hard-coded. If we see ember-cli-typescript, ts files are
-    // resolvable. Once we implement a preprocessor-registration build hook,
-    // this logic can be pushed down first into `@embroider/compat` (which can
-    // generate the appropriate hooks when it upcompiles ember-cli-typescript),
-    // and then later into ember-cli-typescript itself (which can ship a v2
-    // version with the hook).
     //
-    // Typescript is a slightly weird example of a preprocessor because it gets
-    // implemented in babel, so all we realy need to do is make the extension
-    // resolvable and there's no other "loader" or anything to apply.
-    if (this.activeAddonChildren().find(pkg => pkg.name === 'ember-cli-typescript')) {
-      extensions.unshift('.ts');
-    }
-
-    return extensions;
+    // For TS, we defer to ember-cli-babel, and the setting for
+    // "enableTypescriptTransform" can be set with and without
+    // ember-cli-typescript
+    return ['.wasm', '.mjs', '.js', '.json', '.hbs', '.ts'];
```

Comment:
This introduces a bug that causes TypeScript component files to be ignored when `staticComponents: true`

should be: `return ['.ts', '.wasm', '.mjs', '.js', '.json', '.hbs'];`

Reply:
how does that happen / how does moving ts to the front resolve it?

Reply:
I'm not at all familiar with the internals of embroider and had to diagnose this by trial-and-error.

The previous version of the code unshifts '`.ts'` onto the array (placing it at the front). This change moves it to the end of the array.

When I inspected $TMPDIR/embroider/.../components, I found three files for each of my TypeScript components: .hbs, .js, and .ts.

.hbs and .ts were as expected. .js contains a templateOnlyComponent stub.

I can only deduce that in the process of assembling the app, embroider looks for the component implementation file in the order of the extensions in this array, so moving .ts to the end of the array means that it finds the empty stub implementation and ignores the actual implementation in the .ts file.

I have no idea why the stub .js implementations are being generated if a .ts backing class definition exists. Perhaps this is a different bug, or perhaps intended behavior?

The result is a broken component with no backing class.

Doesn't happen if staticComponents = false in the embroider config.